frontend/test/pytest/test_autograph.py

```diff
@@ -1758,5 +1758,83 @@ def g():
         assert g() == 36.4
 
 
+class TestJaxIndexAssignment:
+    """Test Jax index assignment"""
+
+    def test_single_index_assignment_one_item(self):
+        """Test single index assignment for Jax arrays for one array item."""
+
+        @qjit(autograph=True)
+        def zero_last_element_single_assignment_syntax(x):
+            """Set the last element of x to 0 using single index assignment"""
+
+            last_element = x.shape[0] - 1
+            x[last_element] = 0
+            return x
+
+        @qjit(autograph=True)
+        def zero_last_element_at_set_syntax(x):
+            """Set the last element of x to 0 using at and set"""
+
+            last_element = x.shape[0] - 1
+            x = x.at[last_element].set(0)
+            return x
+
+        @qjit(autograph=True)
+        def zero_last_element_python_array(x):
+            """Set the last element of a python array to 0"""
+
+            last_element = len(x) - 1
+            x[last_element] = 0
+            return jnp.array(x)
```

Comment:
What's the difference to case 1? `x` will always be an array when it is an argument of a compiled function.

Reply:
I believe that should be a different test function then? Generally one test function should be testing one "logic path".

Reply:
Although besides the above comment, my point is that `x` is still a JAX array in your example. Why would it be something else?

Reply:
Done!